integrations/dav-server-fs-opendal/src/webdav_dir_entry.rs

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use dav_server::fs::DavDirEntry;
+use futures::FutureExt;
+use opendal::Entry;
+use opendal::Operator;
+
+use super::webdav_file::convert_error;
+use super::webdav_metadata::WebdavMetaData;
+
+pub struct WebDAVDirEntry {
```

Comment:
All public types in crate must have comments.

Reply:
will add at next PR